src/site/stages/build/process-cms-exports/findMatchingEntities.js

```diff
@@ -0,0 +1,72 @@
+const assert = require('assert');
+const fs = require('fs');
+const assembleEntityTreeFactory = require('./index');
+const { readEntity } = require('./helpers');
+
+/*
+ * NOTE: This would ideally be in transformers/helpers.js but ./index.js depends
+ * on that file already, so it would create a circular dependency.
+ */
+
+/**
+ * Search for all entities matching the parameters and return the transformed
+ * entities.
+ *
+ * @param {String} baseType - The base type of the entity
+ * @param {String} contentDir - The path to the directory to read the files from
+ * @param {String} subType - [Optional] The subType of the entity
+ * @param {Function} filter - [Optional] A function which recieves the object
+ *                              read from the JSON file and returns true if we
+ *                              should keep the entity or false if we should not
+ *
+ * TODO: Memoize this function to improve speed if the build is slow because
+ * of this CMS content transformation process.
+ *
+ * @return {Array<Object>} - The transformed entities whose raw form matches
+ *                           the parameters
+ */
+function findMatchingEntities(baseType, contentDir, { subType, filter } = {}) {
+  // Sanity checks
+  assert(
+    typeof baseType === 'string',
+    `baseType needs to be a string. Found ${typeof baseType} (${baseType})`,
+  );
+  assert(
+    fs.lstatSync(contentDir).isDirectory(),
+    `${contentDir} is not a directory.`,
+  );
+  if (subType)
+    assert(
+      typeof subType === 'string',
+      `subType needs to be a string. Found ${typeof subType} (${subType})`,
+    );
+  if (filter)
+    assert(
+      typeof filter === 'function',
+      `filter needs to be a string. Found ${typeof filter} (${filter})`,
+    );
+
+  const assembleEntityTree = assembleEntityTreeFactory(contentDir);
+
+  // Look through contentDir for all `${baseType}.*.json` files
+  return (
+    fs
+      .readdirSync(contentDir)
+      .filter(name => name.startsWith(`${baseType}.`))
+      .map(name => {
+        const uuid = name.split('.')[1];
+        return readEntity(contentDir, baseType, uuid, { noLog: true });
+      })
+      .filter(
+        // Filter them by `subType` if available
+        // NOTE: Not all content models keep their subType in `.type[0]`
+        // If we have to search for those content models, we'll need to update this
+        entity => (subType ? entity.type[0].target_id === subType : true),
+      )
+      // Filter them by `filter` if available
+      .filter(entity => (filter ? filter(entity) : true))
+      .map(entity => assembleEntityTree(entity))
```

Comment:
If you wanted I think you could just pass the function name directly:

```js
       .map(assembleEntityTree) 
```

Of course, this is under the assumption that it is obvious what each step outputs and what input is expected.  Maybe there is value in pointing out that we are dealing with an `entity`, and that `assembleEntityTree` is a function that expects an `entity` as an argument.

Reply:
`assembleEntityTree` actually has [two other optional parameters](https://github.com/department-of-veterans-affairs/vets-website/blob/findMatchingEntities-cv/src/site/stages/build/process-cms-exports/index.js#L33), so if we did `.map(assembleEntityTree)`, [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) will pass `index` as the `ancestors` and `array` as the `parentFieldName`. It'd break things.

Reply:
Good to know!